dsp-shared/src/main/scala/dsp/valueobjects/Schema.scala

```diff
@@ -0,0 +1,209 @@
+            validateGuiObjectsPointingToList(guiElement, guiAttributes).fold(
+              e => Validation.fail(e.head),
+              v => Validation.succeed(v)
+            )
+
+          // gui element is a slider, so it needs two gui attributes min and max
+          case Some(guiElement) if guiElement.value == guiElementSlider =>
+            validateGuiObjectSlider(guiElement, guiAttributes).fold(
+              e => Validation.fail(e.head),
+              v => Validation.succeed(v)
+            )
+
+          case _ =>
+            Validation.fail(
+              BadRequestException(
+                s"Unable to validate gui attributes. Unknown value for salsah-gui:guiElement: $guiElement."
+              )
+            )
+        }
+
+        return validatedGuiAttributes.fold(
+          e => Validation.fail(e.head),
+          v => Validation.succeed(new GuiObject(v, guiElement) {})
+        )
+
+      }
+
+      Validation.succeed(new GuiObject(guiAttributes, guiElement) {})
+    }
+  }
+
+  /**
+   * GuiAttribute value object.
+   *
+   * @param k the key parameter of the gui attribute ('min', 'hlist', 'size' etc.)
+   * @param v the value parameter of the gui attribute (an int, a list's IRI etc.)
+   */
+  sealed abstract case class GuiAttribute private (k: String, v: String) {
+    val value = k + "=" + v
+  }
+  object GuiAttribute {
+    def make(keyValue: String): Validation[Throwable, GuiAttribute] = {
+      val k: String = keyValue.split("=").head.trim()
+      val v: String =
+        keyValue.split("=").last.trim() // TODO also check the type of the value (integer, string etc.)
+
+      if (keyValue.isEmpty) {
+        Validation.fail(BadRequestException(SchemaErrorMessages.GuiAttributeMissing))
+      } else if (!SalsahGui.GuiAttributes.contains(k)) {
+        Validation.fail(BadRequestException(SchemaErrorMessages.GuiAttributeUnknown))
+      } else {
+        Validation.succeed(new GuiAttribute(k, v) {})
+      }
+    }
+
+  }
+
+  /**
+   * GuiElement value object.
+   */
+  sealed abstract case class GuiElement private (value: String)
+  object GuiElement {
+    def make(value: String): Validation[Throwable, GuiElement] =
+      if (value.isEmpty) {
+        Validation.fail(BadRequestException(SchemaErrorMessages.GuiElementMissing))
+      } else if (!SalsahGui.GuiElements.contains(value)) {
+        Validation.fail(BadRequestException(SchemaErrorMessages.GuiElementUnknown))
+      } else {
+        Validation.succeed(new GuiElement(value) {})
+      }
+  }
+
+  /**
+   * Validates if gui elements that require pointing to a list (List, Radio, Pulldown) actually point to a list
+   *
+   * @param guiElement the gui element that needs to be validated
+   * @param guiAttributes the gui attributes that need to be validated
+   *
+   * @return either the validated list of gui attributes or a [[dsp.errors.BadRequestException]]
+   */
+  private[valueobjects] def validateGuiObjectsPointingToList(
```

Comment:
no idea if that is how it should be done... but cool that it works

Reply:
@BalduinLandolt Do you mean the package scoped private method?

Reply:
yes, I didn't know this construct... Scala still baffles me at times - but in a good way!  
The reason I wrote "not sure if that's how it should be done" is that we had a discussion earlier, to which I was referring, where we the three of us were unclear what best practices are regarding unit-testing private methods.